.github/PULL_REQUEST_TEMPLATE/release.md

```diff
@@ -0,0 +1,77 @@
+# Release vFIXME
+
+## Checklist
+
+- [ ] [Create a new github milestone](https://github.com/DataDog/dd-trace-go/milestones/new) for the next version and
+  move any open PRs and issues that are in the current release milestone to the new one.
+
+- [ ] Draft [the release note](https://github.com/DataDog/dd-trace-go/releases/new) and ask the teams for its review on
+  the `#guild-dd-go` channel.
+
+- Deploy a release candidate:
+    - Create a release candidate version on this release branch:
+        - [ ] Bump and commit the release candidate version string in `/internal/version/version.go` (eg. `v1.2.3-rc.4`)
+        - [ ] Tag the bump commit with the release candidate version (eg. `git tag v1.2.3-rc.4`)
+        - [ ] Push the tag and commit (eg. `git push --tags origin release-v1.2.3`)
+    - Deploy some Datadog services to staging:
+        - [ ] Give a heads-up on the channel `#staging-headsup` that you are starting the deployment.
```

Comment:
Can we just post then continue, or should we wait someone to ACK it. Is there a certain time of day we should do this?